src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -818,6 +818,7 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                 } else if (callback instanceof CredentialCallback) {
                     final CredentialCallback credentialCallback = (CredentialCallback) callback;
 
+                    // TODO: should consider realm from RealmCallback
```

Comment:
If we wanted to do this, I think the right place to do this would be in org.wildfly.security.sasl.digest.AbstractDigestMechanism#getPredigestedSaltedPassword.  The problem is that the parameters depend on the selected realm, and the realm callback is handled at the same time as the credential callback.  So, it might not actually be possible.